linalg/simd.hpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
```

Comment:
Why was the motivation to move all the files from `config` to `linalg`?

Reply:
These are not really configuration files. The two logical choices for me are either `linalg` or `general`.